Backend/app/rider-platform/rider-app/Main/src/SharedLogic/Person.hs

```diff
@@ -0,0 +1,106 @@
+{-
+ Copyright 2022-23, Juspay India Pvt Ltd
+
+ This program is free software: you can redistribute it and/or modify it under the terms of the GNU Affero General Public License
+
+ as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version. This program
+
+ is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+
+ or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more details. You should have received a copy of
+
+ the GNU Affero General Public License along with this program. If not, see <https://www.gnu.org/licenses/>.
+-}
+
+module SharedLogic.Person where
+
+import Data.Time hiding (getCurrentTime)
+import Data.Time.Calendar.WeekDate
+import qualified Domain.Types.BookingCancellationReason as DBCR
+import qualified Domain.Types.Merchant as Merchant
+import qualified Domain.Types.Person as DP
+import qualified Domain.Types.Person.PersonStats as DPS
+import qualified Domain.Types.Ride as DR
+import Kernel.Beam.Functions
+import Kernel.Prelude
+import Kernel.Storage.Esqueleto (EsqDBFlow)
+import Kernel.Storage.Esqueleto.Config (EsqDBReplicaFlow)
+import Kernel.Types.Error
+import Kernel.Types.Id
+import Kernel.Utils.Common (fromMaybeM, getCurrentTime)
+import qualified Storage.CachedQueries.CacheConfig as SCC
+import qualified Storage.CachedQueries.Merchant as CQMerchant
+import qualified Storage.Queries.Booking as QBooking
+import qualified Storage.Queries.BookingCancellationReason as QBCR
+import qualified Storage.Queries.Ride as QRide
+import Tools.Metrics (CoreMetrics)
+
+backfillPersonStats :: (EsqDBFlow m r, EsqDBReplicaFlow m r, SCC.CacheFlow m r, CoreMetrics m) => Id DP.Person -> Id Merchant.Merchant -> m DPS.PersonStats
+backfillPersonStats personId merchantId = do
+  cancelledBookingIds <- runInReplica $ QBooking.findAllCancelledBookingIdsByRider personId
+  userCancelledRides <- runInReplica $ QBCR.countCancelledBookingsByBookingIds cancelledBookingIds DBCR.ByUser
+  driverCancelledRides <- runInReplica $ QBCR.countCancelledBookingsByBookingIds cancelledBookingIds DBCR.ByDriver
+  completedRides <- runInReplica QRide.findAllCompletedRides
+  merchant <- CQMerchant.findById merchantId >>= fromMaybeM (MerchantDoesNotExist merchantId.getId)
+  let minuteDiffFromUTC = (merchant.timeDiffFromUtc.getSeconds) `div` 60
+  now <- getCurrentTime
+  let completedRidesCnt = length completedRides
+  let (weekendRides, weekdayRides) = countWeekdaysAndWeekendsRide completedRides minuteDiffFromUTC
+      eveningPeakRides = getWeekdayEveningPeakRides completedRides minuteDiffFromUTC
+      morningPeakRides = getWeekdayMorningPeakRides completedRides minuteDiffFromUTC
+      offPeakRides = completedRidesCnt - (morningPeakRides + eveningPeakRides + weekendRides)
```

Comment:
`weekendPeakRides` should be here instead of `weekendRides`

Reply:
they are not peak rides, they are the weekend and weekday rides

Reply:
modified